src/scheduler/resv_info.c

```diff
@@ -231,6 +220,28 @@ query_reservations(server_info *sinfo, struct batch_status *resvs)
 			continue;
 		}
 
+		/* Make sure it is not a future reservation that is being deleted, if so ignore it */
+		if (resresv->resv->resv_state == RESV_BEING_DELETED && resresv->start > sinfo->server_time) {
```

Comment:
one question: if a reservation is in RESV_BEING_DELETED, is it for certain that the server has not freed the resources on the vnodes yet? for instance, If a user deletes a previously running reservation so that it's in RESV_BEING_DELETED state when the scheduler queries it, is it possible that the reservation's vnodes have been free'd even though it is in state RESV_BEING_DELETED? If yes, then this we can fall into the same trap even with this check right?

Reply:
Server Ideally should not free resources unless the reservation has been completely deleted. In case we are also safe because the reservation which is being deleted has not yet started, so no resources would be assigned to the reservation and scheduler does not really care about a future reservation that is being deleted.

Reply:
@agrawalravi90 I now get the problem you are stating, thanks for explaining it. I'll fix it